### PR TITLE
[cssom] Add tests for the inline CSSStyleDeclaration being mutated not removing the style attribute.

### PR DESCRIPTION
**Pull request by emilio**
This adds tests for https://github.com/whatwg/html/issues/2306.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5392)

### CODE REVIEW

**Comment by emilio** [comment]
@annevk or @zcorpan, could you take a look, thanks!

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/219098256)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 595cfe500bfbc1e9b16f34f8ab435b12a129b1de
Using browser at version BuildID 20170405115424; SourceStamp b043233ec04f06768d59dcdfb9e928142280f3cc
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5392/css/cssom-1/css-style-attribute-modifications.html">/css/cssom-1/css-style-attribute-modifications.html</a></summary>


|                               Subtest                               | Results | Messages |
|---------------------------------------------------------------------|---------|----------|
|                                                                     | OK      |          |
| `Mutating the style declaration doens't remove the style attribute` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/219098257)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 595cfe500bfbc1e9b16f34f8ab435b12a129b1de
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5392/css/cssom-1/css-style-attribute-modifications.html">/css/cssom-1/css-style-attribute-modifications.html</a></summary>


|                               Subtest                               | Results |                Messages                |
|---------------------------------------------------------------------|---------|----------------------------------------|
|                                                                     | OK      |                                        |
| `Mutating the style declaration doens't remove the style attribute` | FAIL    | `assert_true: expected true got false` |

</details>
</details>

**Comment by foolip** [comment]
Can you put this in cssom/ instead, matching https://github.com/w3c/web-platform-tests/pull/5269?

File: css/cssom-1/css-style-attribute-modifications.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<link rel="help" href="https://drafts.csswg.org/cssom/#the-elementcssinlinestyle-interface">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="test" style="color: red"></div>
+<script>
+test(function() {
+  var el = document.getElementById("test");
+  el.style.color = "";
+  assert_true(el.hasAttribute("style"));
+}, "Mutating the style declaration doens't remove the style attribute");
```
Comment (gsnedders) [inline css/cssom-1/css-style-attribute-modifications.html]: *doesn't

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5392/)

**Comment by emilio** [comment]
> Can you put this in cssom/ instead, matching #5269?

Done :)

File: css/cssom-1/css-style-attribute-modifications.html (hunk 1 above)
Comment (emilio) [inline css/cssom-1/css-style-attribute-modifications.html]: Good catch, thanks :)

**Review by foolip — APPROVED** [review]

**Pull request merged**